circuit/src/tables.rs

```diff
@@ -174,14 +214,28 @@ impl<F: CircuitField> CircuitRunner<F> {
             ) => {
                 // Type match - good!
             }
+            (crate::op::NonPrimitiveOp::HashAbsorb { .. }, _) => {
+                panic!("HashAbsorb operation does not take private data");
```

Comment:
you could add the result of the operation in the private data. In this way you can remove the need of passing the permutation to the runner.